sdks/python/apache_beam/runners/worker/worker_pool_main.py

```diff
@@ -0,0 +1,150 @@
+    try:
+      if self._use_process:
+        command = ['python', '-c',
+                   'from apache_beam.runners.worker.sdk_worker '
+                   'import SdkHarness; '
+                   'SdkHarness("%s",worker_count=%d,worker_id="%s").run()' % (
+                       start_worker_request.control_endpoint.url,
+                       self._worker_threads,
+                       start_worker_request.worker_id)]
+        if self._container_executable:
+          # command as per container spec
+          command = [self._container_executable,
+                     '--id=%s' % start_worker_request.worker_id,
+                     '--logging_endpoint=%s'
+                     % start_worker_request.logging_endpoint.url,
+                     '--artifact_endpoint=%s'
+                     % start_worker_request.artifact_endpoint.url,
+                     '--provision_endpoint=%s'
+                     % start_worker_request.provision_endpoint.url,
+                     '--control_endpoint=%s'
+                     % start_worker_request.control_endpoint.url,
+                    ]
+
+        logging.warn("Starting worker with command %s" % command)
+        worker_process = subprocess.Popen(command, stdout=subprocess.PIPE)
+        self._worker_processes[start_worker_request.worker_id] = worker_process
+
+        # Register to kill the subprocess on exit.
+        atexit.register(worker_process.kill)
+      else:
+        worker = sdk_worker.SdkHarness(
+            start_worker_request.control_endpoint.url,
+            worker_count=self._worker_threads,
+            worker_id=start_worker_request.worker_id)
+        worker_thread = threading.Thread(
+            name='run_worker_%s' % start_worker_request.worker_id,
+            target=worker.run)
+        worker_thread.daemon = True
+        worker_thread.start()
+
+      return beam_fn_api_pb2.StartWorkerResponse()
+    except Exception as exn:
+      return beam_fn_api_pb2.StartWorkerResponse(error=str(exn))
+
+  def StopWorker(self, stop_worker_request, unused_context):
+    worker_process = self._worker_processes.pop(stop_worker_request.worker_id)
+    if worker_process:
+      logging.info("Stopping worker %s" % stop_worker_request.worker_id)
+      threading.Timer(1, lambda: worker_process.kill()).start()
+      worker_process.communicate()
+    return beam_fn_api_pb2.StartWorkerResponse()
+
+
+def main(argv=None):
+  """Entry point for worker pool service for external environments."""
+
+  parser = argparse.ArgumentParser()
+  parser.add_argument('--worker_threads',
+                      type=int,
+                      default=12,
+                      help='Number of threads per SDK worker.')
+  parser.add_argument('--container_executable',
+                      type=str,
+                      default=None,
+                      help='Executable that implements the container contract.')
+  parser.add_argument('--servicer_port',
```

Comment:
```suggestion
  parser.add_argument('--service_port',
```
This name looks more intuitive to me.